bot-lib/src/commands/mod.rs

```diff
@@ -6,3 +6,24 @@ pub mod lynch;
 pub mod register;
 pub mod remove_bot_role;
 pub mod timeout;
+pub mod reset_class_categories;
+pub mod delete_class_category;
+pub mod class_roles;
+
+use color_eyre::eyre::Result;
+use poise::serenity_prelude::{GuildChannel, GuildId};
+use regex::Regex;
+use crate::data::PoiseContext;
+
+/// Finds all channels in the given guild, where the name matches the given regex
+pub async fn find_channels(ctx: PoiseContext<'_>, guild: GuildId, pattern: Regex) -> Result<Vec<GuildChannel>>{
+    let channels = guild.channels(ctx).await?;
```

Comment:
Is this cached? If not, this could be very expensive since `reset_class_categories` will end up calling this many times.

Reply:
I believe the channels are cached by poise, most other things are to a certain extent. Discord API calls are going to be a bigger bottleneck anyway, so I'm not too worried (plus this is a once-a-semester command). When the database is ready (eventually), I want to change this stuff over to just link together classes and channels directly, which solves this potential runtime issue and opens up the option for more flexibility. The config file additions were supposed to do this, but in my testing the data gets way too complicated way too quickly, and storing dynamic data in the config is probably a bad idea.